### PR TITLE
chore(context): changed 'context' command to work in batch

### PR DESCRIPTION
**Pull request by YuryShkoda**

### CODE REVIEW

**Review by medjedovicm — APPROVED** [review]

File: src/sasjs-help/index.js [hunk 1]
```diff
@@ -101,18 +101,15 @@ export async function printHelpText() {
...
+          example: sasjs create myTarget -source ../contextConfig.json
         * ${chalk.cyanBright('edit')} - edits existing context.
-          1) Provide target name.
-          2) Provide path to context config file (eg. ./contextConfig.json).
-        * ${chalk.cyanBright('remove')} - removes existing context.
-          1) Provide target name.
-          2) Provide path to context config file (eg. ./contextConfig.json).
+          example: sasjs edit myTarget -source ../contextConfig.json
+        * ${chalk.cyanBright('delete')} - deletes existing context.
+          example: sasjs delete myTarget -source ../contextConfig.json
```
Comment (allanbowe) [inline src/sasjs-help/index.js:108]: why is the `-source` parameter needed here?

also, can we provide an example of the required JSON structure in the help?

File: src/sasjs-help/index.js [hunk 2]
```diff
@@ -101,18 +101,15 @@ export async function printHelpText() {
       'context <command>'
     )} - performs operations on contexts.
         * ${chalk.cyanBright('create')} - creates new context.
-          1) Provide target name.
-          2) Provide path to context config file (eg. ./contextConfig.json).
+          example: sasjs create myTarget -source ../contextConfig.json
```
Comment (allanbowe) [inline src/sasjs-help/index.js:104]: myTarget is a misleading context name - in SASjs nomenclature, a target is a SAS server.  

which is a good point, we should add a target parameter, and default to the first target if a target is not found in the sasjsconfig file

**Review by allanbowe — APPROVED** [review]

**Comment by allanbowe** [comment]
Didn't mean to hit approve - the documentation needs improving, also the part about the target needs clarifying.

Finally - has this been tested with a global (not just local) config?

File: src/sasjs-help/index.js (hunk 1 above)
Comment (YuryShkoda) [reply src/sasjs-help/index.js]: fixed

File: src/sasjs-help/index.js (hunk 2 above)
Comment (YuryShkoda) [reply src/sasjs-help/index.js]: fixed

File: src/sasjs-help/index.js [hunk 3]
```diff
@@ -101,18 +101,38 @@ export async function printHelpText() {
...
+
+          context config example:
+            {
+              "name": "New context",
+              "launchName": "SAS Job Execution launcher context",
+              "sharedAccountId": "cas",
+              "autoExecLines": ["%put hello;"],
+              "authorizedUsers": []
+            }
+          NOTE: Keys "name", "launchName", "sharedAccountId", "autoExecLines", "authorizedUsers" are required.
```
Comment (allanbowe) [inline src/sasjs-help/index.js:115]: are we sure that autoExecLines is required?

**Review by allanbowe — APPROVED** [review]

File: src/sasjs-context/index.js [hunk 4]
```diff
@@ -16,27 +16,74 @@ export async function processContext(command) {
...
+        ).join('\n')}`
+      )
+    )
 
     return
   }
 
-  const targetName = await getAndValidateTargetName()
+  const commandExample =
+    'sasjs context <command> -source ../contextConfig.json -target targetName'
```
Comment (medjedovicm) [inline src/sasjs-context/index.js:31]: When passing parameters that are more then one letter, should it be prepended with 2 dashes? `--`

File: src/sasjs-help/index.js (hunk 3 above)
Comment (YuryShkoda) [reply src/sasjs-help/index.js:115]: according addapter's function, yes
image

File: src/sasjs-context/index.js (hunk 4 above)
Comment (YuryShkoda) [reply src/sasjs-context/index.js:31]: @allanbowe should I change it to use `--`?

Comment (YuryShkoda) [reply src/sasjs-context/index.js:31]: fixed

**Pull request merged**